packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/internal/SyncSessionImpl.kt

```diff
@@ -73,6 +83,43 @@ internal open class SyncSessionImpl(
         RealmInterop.realm_sync_session_resume(nativePointer)
     }
 
+    @Suppress("invisible_member", "invisible_reference")
+    override fun progress(
+        direction: Direction,
+        progressMode: ProgressMode,
+    ): Flow<Progress> {
+        val callbackFlow: Flow<Progress> = callbackFlow {
+            val token = RealmInterop.realm_sync_session_register_progress_notifier(
+                nativePointer,
+                when (direction) {
+                    Direction.DOWNLOAD -> ProgressDirection.RLM_SYNC_PROGRESS_DIRECTION_DOWNLOAD
+                    Direction.UPLOAD -> ProgressDirection.RLM_SYNC_PROGRESS_DIRECTION_UPLOAD
+                },
+                progressMode == ProgressMode.INDEFINITELY
+            ) { transferredBytes: Long, totalBytes: Long ->
+                val progress = Progress(transferredBytes.toULong(), totalBytes.toULong())
+                trySend(progress)
+                if (progressMode == ProgressMode.CURRENT_CHANGES && progress.isTransferComplete) {
+                    close()
+                }
+            }
+            awaitClose {
+                RealmInterop.realm_sync_session_unregister_progress_notifier(nativePointer, token)
+            }
+        }
+        return callbackFlow
+            .onEach {
+                // If we are done downloading data ensure that the realm is up to date as users
+                // would expect to see all the downloaded data
+                if (it.isTransferComplete) {
+                    if (direction == Direction.DOWNLOAD) {
+                        realm?.refresh()
```

Comment:
Are the downloaded changes integrated by the time the transfer is done, or after? We could end in a race condition where the Realm is not refreshed with the latest changes.

Reply:
There is no guarantee that it is actually downloaded (https://mongodb.slack.com/archives/C010R3CMP3N/p1668006638842349), so I guess the comment is wrong. Since I migrated the test that verifies that objects are actually transferred and that there is no effort to sync it, then I guess it is always the case, but ... 🤷. 

This call are not as brutal as `uploadAllLocalChanges` and `downloadAllServerChanges` and seems work for the test, so seems like a middle way. Should we rather just remove it?

Reply:
Yeah, read the core discussion. That is slightly annoying and something we definitely need to document in the Dokka for the method.

I guess the problem that keeping this might make it seem like it works, until it doesn't :( So it is probably safer to remove it, but I could be convinced either way.

It does feel like something is missing in Core though... I feel there should be one way to tell when data downloaded has been integrated, but that is probably for another PR/discussion.